### PR TITLE
Add version beacon service event model

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Continuing to dissect this PR https://github.com/onflow/flow-go/pull/3736 into smaller chunks.

This just adds the service event model and tests. 

I changed the equal to tests and some naming and comments from the original in the #3736 PR.

### CODE REVIEW

File: utils/unittest/fixtures.go [hunk 1]
```diff
@@ -1901,9 +1970,30 @@ func EpochCommitFixture(opts ...func(*flow.EpochCommit)) *flow.EpochCommit {
 	return commit
 }
 
+func VersionBeaconFixture(
+	opts ...func(beacon *flow.VersionBeacon),
```
Comment (pattyshack) [inline utils/unittest/fixtures.go:1974]: maybe rm opts.  it's unclear to me what it's used for.

File: utils/unittest/fixtures.go [hunk 2]
```diff
@@ -777,11 +802,13 @@ func ServiceEventsFixture(n int) flow.ServiceEventList {
 	sel := make(flow.ServiceEventList, n)
 
 	for ; n > 0; n-- {
```
Comment (pattyshack) [inline utils/unittest/fixtures.go:804]: could you reverse the loop iteration?  `for i := 0; i < n; i++` seems more neutral

File: engine/common/rpc/convert/convert.go [hunk 3]
```diff
@@ -813,6 +837,13 @@ func MessageToServiceEvent(m *entities.ServiceEvent) (*flow.ServiceEvent, error)
 			return nil, fmt.Errorf("failed to marshal to EpochCommit event: %w", err)
 		}
 		event = commit
+	case flow.ServiceEventVersionBeacon:
+		versionBeacon := new(flow.VersionBeacon)
+		err := json.Unmarshal(rawEvent, versionBeacon)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal to VersionBeacon event: %w", err)
```
Comment (pattyshack) [inline engine/common/rpc/convert/convert.go:844]: maybe dedup the Unmarshal calls.

e.g.,
```
var event := interface{}
switch m.Type {
case flow.ServiceEventSetup:
  event = new(flow.EpochSetup)
case flow.ServiceEventCommit:
  event = new(flow.EpochCommit)
case flow.ServiceEventVersionBeacon:
  event = new(flow.VersionBeacon)
default:
  return ...
}

err := json.Unmarshal(m.Payload, event)
if err != nil {
  return nil, fmt.Error("... %T event: %w", event, err)
}

return &flow.ServiceEvent{...}
```

File: model/flow/service_event.go [hunk 4]
```diff
@@ -87,6 +88,13 @@ func (se *ServiceEvent) UnmarshalJSON(b []byte) error {
 			return err
 		}
 		event = commit
+	case ServiceEventVersionBeacon:
```
Comment (pattyshack) [inline model/flow/service_event.go:91]: this is a basically a duplicate of the code in convert.go.  maybe just have this call MessageToServiceEvent instead? (same below)

Comment (pattyshack) [reply model/flow/service_event.go:91]: hmm, looks like you can't use MessageToServiceEvent directly.  If you refactor the marshal / unmarshal functions into an interface object, then these become identical code.

File: engine/common/rpc/convert/convert.go (hunk 3 above)
Comment (janezpodhostnik) [reply engine/common/rpc/convert/convert.go:844]: that's a slightly bigger unrelated refactor I put it here https://github.com/onflow/flow-go/pull/4220

File: model/flow/service_event.go (hunk 4 above)
Comment (janezpodhostnik) [reply model/flow/service_event.go:91]: https://github.com/onflow/flow-go/pull/4220

**Review by pattyshack — APPROVED** [review]

**Review by peterargue — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**